src/nomad_material_processing/general/schema.py

```diff
@@ -0,0 +1,278 @@
+    Datetime,
+    Section,
+    Category,
+    MEnum,
+    Reference,
+)
+
+from nomad.datamodel.metainfo.basesections import (
+    CompositeSystem,
+    Component,
+    System,
+    Activity,
+    ActivityStep,
+    ProcessStep,
+    Process,
+    CompositeSystemReference,
+    PureSubstanceSection,
+    PureSubstanceComponent,
+)
+from nomad.config import config
+from nomad.datamodel.metainfo.annotations import (
+    ELNAnnotation,
+    ELNComponentEnum,
+)
+
+from nomad_material_processing import (
+    SubstrateReference,
+    Geometry,
+)
+
+
+class Recipe(ArchiveSection):
+    """
+    A Recipe for a material processing experiment.
+
+    This class will be subclassed for each process that needs a recipe.
+
+    The subclass will inherit Recipe and a specific Process class.
+
+    The only difference between the Recipe and the actual Process is that
+    the datetime and the input samples Entities are hidden in the Recipe.
+    """
+
+    pass
+
+
+class EtchingStep(ActivityStep):
+    """
+    A step of etching process.
+    """
+
+    m_def = Section()
+    duration = Quantity(
+        type=np.float64,
+        description='The elapsed time since the annealing process started.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.NumberEditQuantity, defaultDisplayUnit='minute'
+        ),
+        unit='second',
+    )
+    temperature = Quantity(
+        type=np.float64,
+        description='The temperature of the etching process.',
+        a_eln={'component': 'NumberEditQuantity', 'defaultDisplayUnit': 'celsius'},
+        unit='celsius',
+    )
+    agitation = Quantity(
+        type=MEnum(
+            'Magnetic Stirring',
+            'Sonication',
+        ),
+        description='The agitation method used during the etching process.',
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.EnumEditQuantity,
+        ),
+    )
+    etching_reagents = SubSection(section_def=CompositeSystem, repeats=True)
+
+
+class Etching(Process, EntryData):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section(
+        a_eln=None,
+    )
+    recipe = Quantity(
+        type=Recipe,
+        description=""" The recipe used for the process. If a recipe is found,
+           all the data is copied from the Recipe within the Process.
+           """,
+        a_eln=ELNAnnotation(
+            component=ELNComponentEnum.ReferenceEditQuantity,
+        ),
+    )
+    steps = SubSection(
+        description="""
+        The steps of the etching process.
+        """,
+        section_def=EtchingStep,
+        repeats=True,
+    )
+
+
+class EtchingRecipe(Etching, Recipe, EntryData):
+    """
+    A Recipe for an etching process.
+    """
+
+    m_def = Section(
+        a_eln={'hide': ['datetime', 'samples']},
+    )
+
+
+class Annealing(Process, EntryData):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section(
+        a_eln={'hide': ['steps']},
```

Comment:
Why are we hiding the steps here in the general one? I think annealing is a classic example where you have steps of certain temperatures.

Reply:
solved